### PR TITLE
Use 0.6.8 features

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: deposit_contract.sol [hunk 1]
```diff
@@ -87,7 +87,7 @@ contract DepositContract is IDepositContract {
         require(msg.value >= MIN_DEPOSIT_AMOUNT);
         require(msg.value % GWEI == 0);
         uint deposit_amount = msg.value / GWEI;
-        require(deposit_amount < 2**64);
+        require(deposit_amount < type(uint64).max);
```
Comment (MrChico) [inline deposit_contract.sol]: technically I guess this should be `require(deposit_amount <= type(uint64).max); `

Comment (axic) [reply deposit_contract.sol]: This is an outdated branch, will fix it once the release is out and will also set the pragma fixed to that version.

File: deposit_contract.sol [hunk 2]
```diff
@@ -102,8 +102,8 @@ contract DepositContract is IDepositContract {
         // Compute deposit data root (`DepositData` hash tree root)
         bytes32 pubkey_root = sha256(abi.encodePacked(pubkey, bytes16(0)));
         bytes32 signature_root = sha256(abi.encodePacked(
-            sha256(abi.encodePacked(bytes(signature[:64]))),
-            sha256(abi.encodePacked(bytes(signature[64:]), bytes32(0)))
+            sha256(abi.encodePacked(signature[:64])),
```
Comment (chriseth) [inline deposit_contract.sol:116]: Does this make a difference in the asm?

Comment (axic) [reply deposit_contract.sol:116]: Last I checked it didn't.

File: deposit_contract.sol [hunk 3]
```diff
@@ -87,7 +87,7 @@ contract DepositContract is IDepositContract {
         require(msg.value >= MIN_DEPOSIT_AMOUNT, "DepositContract: deposit value too low");
         require(msg.value % GWEI == 0, "DepositContract: deposit value not multiple of gwei");
         uint deposit_amount = msg.value / GWEI;
-        require(deposit_amount < 2**64, "DepositContract: deposit value too high");
+        require(deposit_amount <= type(uint64).max, "DepositContract: deposit value too high");
```
Comment (leonardoalt) [inline deposit_contract.sol:101]: Nice!

**Review by MrChico — APPROVED** [review]

**Pull request merged**